src/c++/uds/kernelLinux/tests/getTestIndexNames.c

```diff
@@ -30,21 +36,28 @@ enum { BLK_FMODE = FMODE_READ | FMODE_WRITE };
 static struct block_device *get_device_from_name(const char *name)
 {
 	struct block_device *bdev;
+
 #ifdef VDO_USE_ALTERNATE
 #ifdef VDO_USE_ALTERNATE_2
+#ifdef VDO_USE_ALTERNATE_3
 
 	bdev = blkdev_get_by_path(name, BLK_FMODE, NULL);
 #else
 	const struct blk_holder_ops hops = { NULL };
 
 	bdev = blkdev_get_by_path(name, BLK_FMODE, NULL, &hops);
-#endif /* VDO_USE_ALTERNATE_2 */
+#endif /* VDO_USE_ALTERNATE_3 */
 #else
 	const struct blk_holder_ops hops = { NULL };
 	struct bdev_handle *bdev_handle;
 
 	bdev_handle = bdev_open_by_path(name, BLK_FMODE, NULL, &hops);
 	bdev = bdev_handle->bdev;
+#endif /* VDO_USE_ALTERNATE_2 */
+#else
+	const struct blk_holder_ops hops = { NULL };
+
+	bdev = file_bdev(bdev_file_open_by_path(name, BLK_FMODE, NULL, &hops));
 #endif /* VDO_USE_ALTERNATE */
```

Comment:
Looking at the way cleanup happens, I think we need to do something a little subtler here.
I think after we call file_bdev(), we should call get_device(bdev) to get a reference in the block device, then close the struct file object immediately.

We will still have a reference to the bdev (because of get_device()), but not to the file object.  Then we can release the device reference later with put_device().